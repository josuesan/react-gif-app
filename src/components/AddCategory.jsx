import React, { useState } from 'react'
import propTypes from 'prop-types';
const AddCategory = ({ setCategories }) => {
  const [category, setCategory] = useState('');

  const handleCategory = (e) => {
    setCategory(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if (category.trim().length > 2) {
      setCategories(prevState => [
        category,
        ...prevState
      ]);
      setCategory('');
    }
  }
  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input
        type="text"
        onChange={handleCategory}
        value={category}
        placeholder="Add a Category"
      />
    </form>
  )
};

AddCategory.propTypes = {
  setCategories: propTypes.func.isRequired
}
export default AddCategory;