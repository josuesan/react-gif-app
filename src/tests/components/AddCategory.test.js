import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Test to Add Category', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
    jest.clearAllMocks();
  });

  test('should be rendered', async () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the text field', () => {
    const input = wrapper.find('input');
    const value = 'Hi';
    input.simulate('change', {
      target: {
        value
      }
    });
    expect(wrapper.find('input').text().trim()).toBe('');
  })

  test('shouldnt post info on submit ', () => {
    wrapper.find('form').simulate('submit', {
      preventDefault() { },
    });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call submit and clean the text field', () => {
    const value = 'Black Clover';
    const input = wrapper.find('input');
    wrapper.find('input').simulate('change', { target: { value } })
    wrapper.find('form').simulate('submit', {
      preventDefault() { },
    });
    expect(setCategories).toHaveBeenCalled();
    expect(input.prop('value')).toBe('');
  })

});