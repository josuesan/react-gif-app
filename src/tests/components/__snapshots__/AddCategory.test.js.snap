// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to Add Category should be rendered 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Add Category
  </h2>
  <input
    onChange={[Function]}
    placeholder="Add a Category"
    type="text"
    value=""
  />
</form>
`;
